src/dialects/mssql/query-interface.js

```diff
@@ -50,6 +50,29 @@ export class MsSqlQueryInterface extends AbstractQueryInterface {
     return this.sequelize.queryRaw(removeSql, options);
   }
 
+  /**
+    * @override
+    */
+  async bulkInsert(tableName, records, options, attributes) {
+    options = { ...options, type: QueryTypes.INSERT };
+
+    // If more than 1,000 rows are inserted outside of a transaction, we can't guarantee safe rollbacks.
+    // See https://github.com/sequelize/sequelize/issues/15426
+    if ((records.length <= 1000) || options.transaction) {
```

Comment:
I prefer this condition to be inverted so further conditions don't cause too much nesting

Parenthesis around the first part of the condition are redundant

The rest of this method doesn't need to be duplicated, and could call `super.bulkInsert`